views/forms/UserProfileForm/index.jsx

```diff
@@ -17,99 +19,63 @@ const schema = yup.object({
 })
 
 const UserProfileForm = ({ initialValues, onSubmit }) => {
-  const profileImageRef = useRef()
-  const {
-    handleSubmit,
-    control,
-    formState: { errors },
-    watch,
-  } = useForm({
+  const { handleSubmit, control, watch, setValue } = useForm({
     defaultValues: initialValues,
     resolver: yupResolver(schema),
+    mode: 'onChange',
   })
+  const fileInput = useRef()
+  const iconFile = watch('iconFile')
+  const [iconSrc, setIconSrc] = useState()
+  const [iconName, setIconName] = useState()
 
+  useEffect(() => {
+    if (iconFile) {
+      setIconName(iconFile.name)
+      FileModel.toDataURL(iconFile).then(setIconSrc)
+    } else {
+      setIconSrc(undefined)
+    }
+  }, [iconFile])
+
+  const resetIconFileName = () => {
+    setValue('iconFileName', '')
+    setIconName('')
+  }
   return (
     <>
       <form onSubmit={handleSubmit(onSubmit)}>
-        <Controller
+        <UserProfileImage
           control={control}
-          name="icon"
-          render={({ field }) => {
-            return (
-              <div>
-                <input
-                  accept="image/*"
-                  data-testid="icon-input"
-                  id="icon"
-                  multiple
-                  type="file"
-                  {...field}
-                  value={field.value?.fileName}
-                  onChange={(event) => {
-                    const { files } = event.target
-
-                    if (files[0]) {
-                      const reader = new FileReader()
+          iconFileName={iconName}
+          fileInputRef={fileInput}
+          onRemoveIcon={() => {
+            setValue('iconFile', null)
```

Comment:
why not put this and the `fileInput.current.value = ''` line in the `resetIconFileName` function? Also might be helpful to rename that function to this prop name (`onRemoveIcon`) to make it more obvious what its intent is.

Reply:
[2b10634](https://github.com/AMP-SCZ/dpdash/pull/478/commits/2b10634c506de607903b3c18f8cbfd56e8904186)